3rdparty/mbedtls/mbedtls/library/x509write_crt.c

```diff
@@ -308,7 +308,7 @@ int mbedtls_x509write_crt_der( mbedtls_x509write_cert *ctx, unsigned char *buf,
     unsigned char *c, *c2;
     unsigned char hash[64];
     unsigned char sig[MBEDTLS_MPI_MAX_SIZE];
-    unsigned char tmp_buf[2048];
+    unsigned char tmp_buf[2048*4];
```

Comment:
Allocating 8 K variable on the stack would increase the risk of stack overflow.
Is this a good crypto practice?
Is that how OpenSSL handles large data?

Reply:
Yes stack space can be a problem. Doing a malloc is safer than consuming lots of stack space.

Reply:
replaced with dynamic memory allocation.